xamarin/Xamarin.Forms
# Fixes Issue #6187 - Visual Material Underline color should match placeholder color

### Description of Change ###

- Replaces TextColor with Placeholdercolor for setting the underline color in the relavant Material renderers (or base classes if appropriate)

No automated UI test as I do not know how to check the underline color of an Entry using UITest.

### Issues Resolved ### 
- fixes #6187

### API Changes ###

 None

### Platforms Affected ### 

- iOS
- Android
- Tizen

### Behavioral/Visual Changes ###
- Underline color in a Material Entry will now match the PlaceholderColor rather than the TextColor

None

### Before/After Screenshots ### 

![Unfixed_Screenshot](https://user-images.githubusercontent.com/4606509/68979826-bcb07a00-07a2-11ea-9ab2-c9fa2a6dfee3.png)
![Fixed_Screenshot](https://user-images.githubusercontent.com/4606509/68979829-c0440100-07a2-11ea-8213-611555b7f743.png)


### Testing Procedure ###

1. Create a Material Entry control and set the TextColor and the PlaceholderColor. 
2. Make sure the underline color in the Entry matches the PlaceholderColor

### PR Checklist ###

- [*] Rebased on top of the target branch at time of PR
- [*] Changes adhere to coding standard
